internal/grpc/services/gateway/storageprovider.go

```diff
@@ -308,12 +317,92 @@ func (s *svc) DeleteStorageSpace(ctx context.Context, req *provider.DeleteStorag
 
 	res, err := c.DeleteStorageSpace(ctx, req)
 	if err != nil {
-		return &provider.DeleteStorageSpaceResponse{
-			Status: status.NewStatusFromErrType(ctx, "gateway could not call DeleteStorageSpace", err),
-		}, nil
+		code := gstatus.Code(err)
+		if storageid == utils.ShareStorageProviderID && code == codes.Unimplemented {
+			res = &provider.DeleteStorageSpaceResponse{
+				Status: status.NewOK(ctx),
```

Comment:
I'm explicitly checking if it's the ShareStorageProvider because it doesn't implement the `DeleteStorageSpace` method.
Not 100% sure if that makes sens though.

Reply:
Mmmh. I wouldn't say so. Either we ignore all `Unimplemented` errors or we return all. Having a special case for the Sharesstorageprovider feels wrong to me.

Maybe we log `Unimplemented` errors but ignore them?

Reply:
We decided to not handle `Unimplemented` differently.